File: Projeto_WindowsForms/Projeto_WindowsForms/DAL/EmpresaDAO.cs
using System.Data.SqlClient;
using Projeto_WindowsForms.Modelo;

namespace Projeto_WindowsForms.DAL
{
    public class EmpresaDAO
    {
        private SqlDataReader dr;
        private readonly Conexao conexao;

        public EmpresaDAO()
        {
            conexao = new Conexao();
        }

        public void cadastrarEmpresa(Empresa empresa)
        {
            var cmd = new SqlCommand
            {
                CommandText = @"insert into empresa (cnpj, razao_social, nome_fantasia, ativo) values (@cnpj, @razaosocial, @nomefantasia, @ativo)"
            };

            cmd.Parameters.AddWithValue("@cnpj", empresa.Cnpj);
            cmd.Parameters.AddWithValue("@razaosocial", empresa.RazaoSocial);
            cmd.Parameters.AddWithValue("@nomefantasia", empresa.NomeFantasia);
            cmd.Parameters.AddWithValue("@ativo", empresa.Ativo);

            try
            {
                cmd.Connection = conexao.conectar();

                cmd.ExecuteNonQuery();

                conexao.desconectar();
            }
            catch (Exception)
            {
                throw;
            }
            finally
            {
                conexao.desconectar();
            }

        }
        public List<Empresa> listarEmpresa()
        {
            var cmd = new SqlCommand
            {
                CommandText = @"SELECT id, cnpj, nome_fantasia, razao_social, ativo FROM empresa WHERE ativo = 1"
            };

            var listaEmpresa = new List<Empresa>();

            try
            {
                cmd.Connection = conexao.conectar();

                dr = cmd.ExecuteReader();

                if (dr.HasRows)
                {
                    while (dr.Read())
                    {
                        var empresa = new Empresa
                        {
                            Id = int.Parse(dr["id"].ToString()),
                            Cnpj = dr["cnpj"].ToString(),
                            NomeFantasia = dr["nome_fantasia"].ToString(),
                            RazaoSocial = dr["razao_social"].ToString(),
                            Ativo = bool.Parse(dr["ativo"].ToString())
                        };

                        listaEmpresa.Add(empresa);
                    }
                }
                dr.Close();
            }
            catch (Exception)
            {
                throw;
            }
            finally
            {
                conexao.desconectar();
            }

            return listaEmpresa;
        }
    }
}

File: Projeto_Android_Studio/MyApplication/app/src/main/java/com/example/myapplication/DAL/EmpresaDAO.java
package com.example.myapplication.DAL;

import com.example.myapplication.modelo.Colaborador;
import com.example.myapplication.modelo.Empresa;
import com.example.myapplication.modelo.enums.TipoCargo;
import com.example.myapplication.modelo.enums.TipoSexo;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;
import java.util.ArrayList;

public class EmpresaDAO {
    Connection connection;

    public ArrayList<Empresa> listarEmpresa() {

        ArrayList<Empresa> listaEmpresa = new ArrayList<>();

        try
        {
            Conexao conexao = new Conexao();
            connection = conexao.Conectar();

            if (connection != null) {
                String query = "SELECT e.id, e.cnpj, e.razao_social, e.nome_fantasia, e.ativo FROM empresa e WHERE e.ativo = 1";
                Statement st = connection.createStatement();

                ResultSet rs = st.executeQuery(query);

                while (rs.next()) {
                    Empresa empresa = new Empresa();
                    empresa.Id = rs.getInt("id_empresa");
                    empresa.Cnpj = rs.getString("cnpj");
                    empresa.RazaoSocial = rs.getString("razao_social");
                    empresa.NomeFantasia = rs.getString("nome_fantasia");
                    empresa.Ativo = Boolean.parseBoolean(rs.getString("e_ativo"));

                    listaEmpresa.add(empresa);
                }
            }
        }
        catch (Exception ex)
        {
            String el = ex.getMessage();
        }

        return listaEmpresa;
    }
}
